### PR TITLE
[pdata] Deprecate New[Slice|Map]FromRaw functions

### PR DESCRIPTION
**Pull request by dmitryax**
Use `New[Slice|Map]().FromRaw()` instead fo consistency with the interface provided by primitive slices.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/6041

### CODE REVIEW

**Comment by dmitryax** [comment]
@bogdandrutu  `NewMapFromRaw` is used pretty extensively for tests in contrib to create map instances inline. This PR introduces significant inconvenience. I'm wondering if we should have some additional helper... I'll submit a draft PR in contrib

**Comment by dmitryax** [comment]
Maybe returning map by `FromRaw` is ok given that we already return map by `Sort`

**Comment by bogdandrutu** [comment]
Another option is to also keep the NewFromRaw as a helper for tests?

File: pdata/pcommon/common_test.go [hunk 1]
```diff
@@ -168,11 +168,11 @@ func TestNilOrigSetValue(t *testing.T) {
 
 	av = NewValueEmpty()
 	av.SetEmptyMapVal().UpsertString("k", "v")
-	assert.Equal(t, NewMapFromRaw(map[string]interface{}{"k": "v"}), av.MapVal())
+	assert.Equal(t, map[string]interface{}{"k": "v"}, av.MapVal().AsRaw())
```
Comment (bogdandrutu) [inline pdata/pcommon/common_test.go:171]:
```golang
	av.SetEmptyMapVal().FromRaw(map[string]interface{}{"k": "v"})
	assert.Equal(t, map[string]interface{}{"k": "v"}, av.MapVal().AsRaw())
```

**Review by bogdandrutu — APPROVED** [review]

File: pdata/pcommon/common_test.go (hunk 1 above)
Comment (dmitryax) [reply pdata/pcommon/common_test.go]: addressed

**Pull request merged**